cmd/thv-operator/pkg/registryapi/config/config.go

```diff
@@ -485,3 +569,178 @@ func buildDatabaseConfig(dbConfig *mcpv1alpha1.MCPRegistryDatabaseConfig) *Datab
 
 	return config
 }
+
+// buildAuthConfig creates an AuthConfig from the CRD spec.
+// If the spec is nil, defaults to anonymous authentication.
+func buildAuthConfig(
+	authConfig *mcpv1alpha1.MCPRegistryAuthConfig,
+) (*AuthConfig, error) {
+	config := &AuthConfig{}
+	if authConfig == nil {
+		// Note: we default to anonymous for backwards compatibility and
+		// because we don't have active production deployments yet.
+		// The plan is to remove this default and require the user to specify
+		// the mode.
+		// TODO: Remove this default once testing is complete.
+		config.Mode = AuthModeAnonymous
+		return config, nil
+	}
+
+	// Map the mode from CRD type to config type
+	switch authConfig.Mode {
+	case mcpv1alpha1.MCPRegistryAuthModeOAuth:
+		config.Mode = AuthModeOAuth
+	case mcpv1alpha1.MCPRegistryAuthModeAnonymous:
+		config.Mode = AuthModeAnonymous
+	default:
+		// Default to anonymous if mode is empty or unrecognized
+		config.Mode = AuthModeAnonymous
```

Comment:
I think the default is fine for now but maybe could be louder (a typo means that you drop to anonymous access...)